cvcl/views/AddUser.py

```diff
@@ -0,0 +1,30 @@
+from django.shortcuts import render
+from django.views.generic import FormView, DetailView, ListView, UpdateView
+from django.http import HttpResponseRedirect
+from django.urls import reverse_lazy
+import csv
+import io
+
+from ..forms import csvForm
+from django.contrib.auth.decorators import user_passes_test
+from django.utils.decorators import method_decorator
+from django import forms
+from ..models import User
+
+
+def is_instructor_check(user):
+    return user.is_instructor
+
+
+@method_decorator(user_passes_test(is_instructor_check), name='dispatch')
```

Comment:
@method_decorator(user_passes_test(is_instructor_check), name='dispatch') .This is what you used. in the course.py file That only checks if a user is an instructor but it doesn't check if the user is the instructor for the specific course